### PR TITLE
LibWeb: Add initial CanvasPath arcTo support

### PR DESCRIPTION
**Pull request by navadev**
This patch adds initial CanvasPath arcTo support for when working with canvas 2D rendering contexts
https://html.spec.whatwg.org/multipage/canvas.html#dom-context-2d-arcto
Example:
```
<canvas style="border: 1px solid black" width=200 height=200 id="canvas"></canvas>
<script>
const canvas = document.getElementById("canvas");
const ctx = canvas.getContext("2d");
ctx.beginPath();
ctx.moveTo(10, 25);
ctx.arcTo(75, 25, 75, 60, 20);
ctx.stroke();
</script>
```
Before:
before_simple

After:
after_simple

Philip html5 canvas test suite (https://philip.html5.org/tests/canvas/suite/tests/index.2d.path.arcTo.html):
Before:
before

After:
after

### CODE REVIEW

File: AK/Math.h [hunk 1]
```diff
@@ -67,11 +67,11 @@ constexpr T to_degrees(T radians)
...
     if (is_constant_evaluated()) {                \
-        if (IsSame<T, long double>)               \
+        if constexpr (IsSame<T, long double>)     \
             return __builtin_##function##l(args); \
-        if (IsSame<T, double>)                    \
+        if constexpr (IsSame<T, double>)          \
             return __builtin_##function(args);    \
-        if (IsSame<T, float>)                     \
+        if constexpr (IsSame<T, float>)           \
             return __builtin_##function##f(args); \
```
Comment (awesomekling) [inline AK/Math.h:75]: This feels like it should be in a separate commit :)

File: Userland/Libraries/LibWeb/HTML/Canvas/CanvasPath.cpp [hunk 2]
```diff
@@ -120,6 +120,80 @@ WebIDL::ExceptionOr<void> CanvasPath::ellipse(float x, float y, float radius_x,
...
 }
 
+// https://html.spec.whatwg.org/multipage/canvas.html#dom-context-2d-arcto
+WebIDL::ExceptionOr<void> CanvasPath::arc_to(double x1, double y1, double x2, double y2, double radius)
+{
+    // 1. If any of the arguments are infinite or NaN, then return.
+    if (isnan(x1) || isnan(y1) || isnan(x2) || isnan(y2) || isnan(y2) || isnan(radius))
+        return {};
+    if (isinf(x1) || isinf(y1) || isinf(x2) || isinf(y2) || isinf(radius))
+        return {};
```
Comment (awesomekling) [inline Userland/Libraries/LibWeb/HTML/Canvas/CanvasPath.cpp:130]: I think we can combine the checks for infinite or NaN into `!isfinite(foo)`

**Review by awesomekling — CHANGES_REQUESTED** [review]
Hello and welcome to the project! Great place to start :)

File: AK/Math.h (hunk 1 above)
Comment (navadev) [reply AK/Math.h]: Hello :) it looks like the analyzer issue this was meant to resolve is no longer happening, and I have no local compile issues, so I've completely removed this from the patch.

File: Userland/Libraries/LibWeb/HTML/Canvas/CanvasPath.cpp (hunk 2 above)
Comment (navadev) [reply Userland/Libraries/LibWeb/HTML/Canvas/CanvasPath.cpp]: Thank you for the suggestion! I've made the update.

**Comment by awesomekling** [comment]
Thanks! It would be cool to incorporate tests for this in our own repo, but I won't block this first PR on that. :)
